app/models/profile.rb

```diff
@@ -40,6 +47,14 @@ def self.ransackable_attributes(_auth_object = nil)
     %w[name]
   end
 
+  def should_generate_new_friendly_id?
+    name_changed? || super
+  end
```

Comment:
This method regenerates the slug when users change their profile name.

Reply:
Lets also add this method in Session model, currently friendly_id stays the same even on title change